src/sys/unix/uds/datagram.rs

```diff
@@ -99,12 +100,10 @@ impl UnixDatagram {
                 sockaddr,
                 socklen,
             );
-            if count > 0 {
-                Ok(1)
-            } else if count == 0 {
-                Ok(0)
-            } else {
-                Err(io::Error::last_os_error())
+            match count.cmp(&0) {
+                Ordering::Greater => Ok(1),
+                Ordering::Equal => Ok(0),
+                Ordering::Less => Err(io::Error::last_os_error()),
```

Comment:
Thoughts on this?

```suggestion
if count == -1 {
    Err(io::Error::last_os_error())
} else {
    Ok(count as i32)
}
```

It's similar to what `syscall!` does under the hood so it may be a more expected handling of the result value (at least when reading the code).

Reply:
I would have done that but the original code returned `Ok(1)` if `count > 0`, that is why I asked on Discord.

Reply:
I've changed it to use the `syscall!` macro if the result doesn't matter.